pg_rsupport.c

```diff
@@ -597,6 +597,7 @@ plr_SPI_execp(SEXP rsaved_plan, SEXP rargvalues)
 	return result;
 }
 
+#if CATALOG_VERSION_NO < 201811201
```

Comment:
This is a compile time check, wouldn't it make more sense to do a runtime check for the version of PostgreSQL

Reply:
It would be somewhat messy to implement [explicit linking](https://docs.microsoft.com/en-us/cpp/build/linking-an-executable-to-a-dll?view=vs-2017#linking-explicitly) with GetProcAddress/dlsym stuff. I think it is easier to call it an ABI change and not try to load plr with the wrong PG version.

Reply:
I was thinking about explicit linking R though... not to deal with adding it to PATH as we can probe the location from windows registry. Using manifests and DLL assembly probing privatePath is not gonna work as we need to go up 3 directories. Delay loading R.DLL is not gonna work as we implicitly link data.. like R_NilValue. Another option would be to split plr.dll into 2 dlls. One would [AddDllDirectory](https://docs.microsoft.com/en-us/windows/desktop/api/libloaderapi/nf-libloaderapi-adddlldirectory) before calling any functions from delay loaded other dll. But it already seems like doesn't worth an effort comparing to changing PATH :-)

Reply:
> What I was contemplating was looking at the server version using sql and then erroring in the method if it was the wrong version.

The plr.dll won't load with the current code as is complaining that SPI_lastoid was not found in postgres.exe

Reply:
ah, ok